model/flow/protocol_state.go

```diff
@@ -59,6 +60,20 @@ type EpochStateContainer struct {
 	// epoch are only allowed to listen to the network but not actively contribute. Such
 	// nodes are _not_ part of `Identities`.
 	ActiveIdentities DynamicIdentityEntryList
+
+	// EpochExtensions represents extensions to the last successful epoch in EFM. In the happy path it is nil.
+	// Epochs in which EFM is triggered will have at least one EpochExtension.
+	// Extensions is an ordered list, such that for each consecutive pair of
+	// EpochExtensions e1, e2, e1.FinalView+1 = e2.FirstView.
```

Comment:
The "last successful Epoch" is not cleanly defined ... at least it conflicts with my intuitive assumption what this term would mean. Assume we had epoch extensions to epoch `N`, and that we have recovered and are now in epoch N+1, with Epoch N+2 already committed. What precisely does "successful" mean here? If it means successfully committed, then that would be epoch N+2. If it means successfully entered, then that "last" epoch would be N+1 ... I hope I manage to explain my concern with this wording. 


```suggestion
	// EpochExtensions contains potential EFM-extensions of this epoch. In the happy path
	// it is nil or empty. An Epoch in which Epoch-Fallback-Mode [EFM] is triggered, will
	// have at least one extension. By convention, the initial extension must satisfy
	//   EpochSetup.FinalView + 1 = EpochExtensions[0].FirstView
	// and each consecutive pair of slice elements must obey
	//   EpochExtensions[i].FinalView+1 = EpochExtensions[i+1].FirstView
```